src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -156,13 +156,45 @@ trait RegularSync {
     }
   }
 
+  /**
+    * Inserts and executes the transactions of the blocks, up to the point to which one of them fails (or we run out of blocks).
+    * If the execution of any block were to fail, newBlocks only contains the NewBlock msgs for all the blocks executed before it.
+    *
+    * @param blocks to execute
+    * @param blockParentTd, td of the parent of the blocks.head block
+    * @param newBlocks which, after adding the corresponding NewBlock msg for blocks, will be broadcasted
+    * @return list of NewBlocks to broadcast (one per block successfully executed) and an error if one happened during execution
+    */
+  @tailrec
+  private def insertBlocks(blocks: Seq[Block], blockParentTd: BigInt,
```

Comment:
btw can we rename this method to `processBlocks` or something similar, it does not only insert (and it does not always insert)